### ISSUE TITLE
Setting StorageLevel to MEMORY_AND_DISK

### ISSUE THREAD

**Issue by ignasva** [DataSystemsLab/GeoSpark#36]
I've noticed that at the moment storage level of SpatialRDDs is hardcoded to MEMORY_ONLY. 

**Question:** Is there a reason for this and are there any potential pitfalls of setting it to MEMORY_AND_DISK_SER?

**Details:**
I'm planning to use GeoSpark to query a large dataset (>30Tb), which does not fit in memory. 

If the storage level is set to MEMORY_ONLY, this would mean that partitions would have to be recomputed as soon as the memory fills up. This is a problem because I'm doing a significant amount of preprocessing before creating SpatialRDDs. This also means that if I have an index set, this index will also have to be recomputed when I run out of memory, which contradicts the whole point of setting an index. Please correct me if I'm wrong.

A (naive) solution to this issue would be simply creating a custom build of GeoSpark where the storage level is replaced to MEMORY_AND_DISK, however, I wanted to make sure that this will not cause any trouble. Ideally, user should be able to set the storage level as a parameter.

Thanks!

**Comment by jiayuasu** [comment]
Hi @ignasva ,

You understand the code correctly. I hardcoded it because spilling onto disk actually is not very time efficient.

To solve this, you don't have to make your own GeoSpark build. Here is an easy workaround (but not perfect):

You can directly use GeoSpark Maven Coordinate. Since you checked the code, you may notice that all SpatialRDD attributes are "public". You can directly call the attribute in your code and set to the persist level you want. For example, 
```
myPointRDD.rawPointRDD.persist(MEMORY_AND_DISK_SER);
```

```
myPointRDD.indexedRDD.persist(MEMORY_AND_DISK_SER);
```

Of course, making your own build may give you the perfect control.

Thanks,
Jia

**Comment by ignasva** [comment]
Hi @jiayuasu,

At first this sounded like a good idea, however, apparently you cannot change the storage level after a Spark RDD is cached. This is the resulting exception:

**java.lang.UnsupportedOperationException**: Cannot change storage level of an RDD after it was already assigned a level

Anyway, it would be awesome to be able to control the storage level in future releases. Keep up the good work!

**Comment by jiayuasu** [comment]
Hi @ignasva ,

Thanks for pointing this out. I should give the granular control back to the user. I will change the cache strategy in GeoSpark.

Thanks,
Jia

**Comment by jiayuasu** [comment]
Solved in GeoSpark 0.4.0
